src/town_cmd.cpp

```diff
@@ -1159,19 +1159,24 @@ static bool GrowTownWithBridge(const Town *t, const TileIndex tile, const DiagDi
 
 	const int delta = TileOffsByDiagDir(bridge_dir);
 
+	/* To prevent really small towns from building disproportionately
+	 * long bridges, make the max a function of its population. */
+	int min_max_bridge_length = 4;
+	int max_bridge_length = 0.001 * t->cache.population + min_max_bridge_length;
```

Comment:
Floating point is a no-no in game code as the rounding might break multi-player. Thankfully, changing it to `/ 1000` is not hard at all.